data/mods/gen9fe/abilities.ts

```diff
@@ -3522,13 +3522,20 @@ export const Abilities: import('../../../sim/dex-abilities').ModdedAbilityDataTa
 		shortDesc: "Secondaries against this Pokemon also proc against the attacker",
 		onModifySecondaries(secondaries) {
 			for (const secondary of secondaries) {
-				secondary.self ||= {};
-				for (const i in secondary) {
-					if (!['self', 'chance'].includes(i)) {
-						const secClone = structuredClone(secondary);
-						secClone.self = undefined;
-						secondary.self = secClone;
-					}
+				const selfEffect: AnyObject = {};
+
+				if (secondary.onHit) selfEffect.onHit = secondary.onHit;
+				if (secondary.boosts) selfEffect.boosts = secondary.boosts;
+				if (secondary.status) selfEffect.status = secondary.status;
+				if (secondary.volatileStatus) selfEffect.volatileStatus = secondary.volatileStatus;
+				if (secondary.sideCondition) selfEffect.sideCondition = secondary.sideCondition;
+				if (secondary.slotCondition) selfEffect.slotCondition = secondary.slotCondition;
+				if (secondary.pseudoWeather) selfEffect.pseudoWeather = secondary.pseudoWeather;
+				if (secondary.terrain) selfEffect.terrain = secondary.terrain;
+				if (secondary.weather) selfEffect.weather = secondary.weather;
```

Comment:
you can probably put all of these properties into a string array and iterate and do something like `if (secondary[property]) selfEffect[i] = secondary[i];`